cppfd/solver.cpp

```diff
@@ -213,6 +238,301 @@ void Solver::solve(stopping_point s_p, linear_solver l_s, adaptative_time_step a
 	      << std::endl;
   }
   timer.reset();
+
+  // Finalize MPI environment
+  #ifdef USE_MPI
+  MPI_Finalize();
+  #endif
+}
+
+void Solver::assemble_parallel_meshes(){
+  uint64_t n_x;
+  uint64_t n_y;
+  uint16_t n_p = this->n_colors_x;
+  uint16_t n_q = this->n_colors_y;
+  double o_x = 0.;
+  double o_y = 0.;
+
+  uint64_t n_cells_p_mesh_x = this->domain_size_x / this->n_p_mesh_x;
+  uint64_t n_cells_p_mesh_y = this->domain_size_y / this->n_p_mesh_y;
+
+  uint64_t remainder_x = this->domain_size_x % n_cells_p_mesh_x;
+  uint64_t remainder_y = this->domain_size_y % n_cells_p_mesh_y;
+  uint64_t r_x;
+  uint64_t r_y = remainder_y;
+
+  int8_t p_mesh_border;
+
+  // case where size of domain is NOT a multiple of the number of parallel meshes given
+  if(remainder_x || remainder_y){
+    // case where there is only one column of parallel meshes
+    if((this->n_p_mesh_x == 1) && (this->n_p_mesh_y != 1)){
+      for(uint64_t ky = 0; ky < this->n_p_mesh_y; ky++){
+        n_y = n_cells_p_mesh_y;
+        // correct parallel mesh size with remainder
+        if(r_y > 0){
+          n_y++;
+          r_y--;
+        }
+
+        // compute location type of current parallel mesh
+        if(ky == 0){
+          p_mesh_border = static_cast<int>(LocationIndexEnum::VERT_BAR_BOT);
```

Comment:
Since `p_mesh_border` is `int8_t`:
```suggestion
          p_mesh_border = static_cast<int8_t>(LocationIndexEnum::VERT_BAR_BOT);
```
here and everywhere below.